src/app/services/baw-api/project/projects.service.ts

```diff
@@ -83,6 +87,36 @@ export class ProjectsService implements StandardApi<Project> {
     url.searchParams.set("selected_timezone_name", selectedTimezone);
     return url.toString();
   }
+
+  public getProjectFor(
+    model: AudioRecording | Site | Region
+  ): Observable<Project[]> {
+    let siteIds: Id[] | undefined;
+
+    if (model instanceof Region) {
+      siteIds = Array.from(model.siteIds);
+    } else if (model instanceof Site) {
+      siteIds = [model.id];
+    } else if (model instanceof AudioRecording) {
+      siteIds = [model.siteId];
+    }
+
+    if (!siteIds) {
+      console.warn("No siteIds found for model", model);
+      return;
```

Comment:
should this return an empty observable? one that is finished?

Reply:
Yeah it should. I'm quite concerned that TypeScript didn't yell at me for this.

This signature of the function doesn't allow `undefined` to be returned

![image](https://github.com/user-attachments/assets/e1525f88-9b29-463b-aa1f-6a659fef6cfb)

Reply:
![image](https://github.com/user-attachments/assets/6b04166e-43c8-4cbf-a032-90b79c12ee63)